airflow/api/auth/backend/kerberos_auth.py

```diff
@@ -141,7 +143,7 @@ def decorated(*args, **kwargs):
             token = "".join(header.split()[1:])
             return_code = _gssapi_authenticate(token)
             if return_code == kerberos.AUTH_GSS_COMPLETE:
-                g.user = ctx.kerberos_user
+                g.user = get_airflow_app().appbuilder.sm.find_user(email=ctx.kerberos_user)
```

Comment:
It works. In case when email domain not equal kerberos realm I suggest to look user by username field. It works too.